### PR TITLE
[REFACTOR]: change spark batch api code to java

### PR DESCRIPTION
**Pull request by huzk8**
change spark batch api code to java

---

## Purpose of this pull request

- [x] seatunnel-spark-api   [batch]

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

**Comment by CalvinKirs** [comment]
hi, can you compile it locally to solve the problem of compilation failure?

**Comment by huzk8** [comment]
I will fix it later



发自我的iPhone


------------------ Original ------------------
From: Kirs ***@***.***&gt;
Date: Mon,Dec 6,2021 11:45 AM
To: InterestingLab/seatunnel ***@***.***&gt;
Cc: huzk ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [InterestingLab/seatunnel] refactor: change spark batch api code to java (PR #742)





 
hi, can you compile it locally to solve the problem of compilation failure?
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by huzk8** [comment]
> hi, can you compile it locally to solve the problem of compilation failure?

**Comment by huzk8** [comment]
@garyelephant @RickyHuo @CalvinKirs Ask help for CR .

**Comment by CalvinKirs** [comment]
Thank you for your contribution. My suggestion is that we keep the Scala code of the plug-in and only modify the kernel module.

**Comment by huzk8** [comment]
Please CR again. @CalvinKirs

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import org.apache.spark.sql.Row;
+
+import java.util.List;
+
+public class SparkBatchExecution implements Execution<BaseSparkBatchSource, BaseSparkTransform, BaseSparkBatchSink> {
+    private SparkEnvironment environment;
+    private Config config = ConfigFactory.empty();
+
+
+
```
Comment (yx91490) [inline seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:39]: duplicated blank line

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    public SparkBatchExecution(SparkEnvironment environment) {
+        this.environment = environment;
+    }
+
+    @Override
+    public void start(List<BaseSparkBatchSource> sources, List<BaseSparkTransform> transforms, List<BaseSparkBatchSink> sinks) {
+        sources.forEach(s -> {
+            BaseSparkSource<Dataset<Row>> ds = s;
```
Comment (yx91490) [inline seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:47]: Local variable 'ds' is redundant

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+    public static void registerTempView(String tableName, Dataset<Row> ds) {
+        ds.createOrReplaceTempView(tableName);
+    }
+
+    public static void registerInputTempView(BaseSparkSource<Dataset<Row>> source, SparkEnvironment environment) {
+        Config conf = source.getConfig();
+        final boolean flag = conf.hasPath(SparkBatchExecution.RESULT_TABLE_NAME);
+
+        if (flag) {
```
Comment (yx91490) [inline seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:95]: better to rename flag to 'needRegister'

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    public static void registerTempView(String tableName, Dataset<Row> ds) {
+        ds.createOrReplaceTempView(tableName);
+    }
+
+    public static void registerInputTempView(BaseSparkSource<Dataset<Row>> source, SparkEnvironment environment) {
+        Config conf = source.getConfig();
+        final boolean flag = conf.hasPath(SparkBatchExecution.RESULT_TABLE_NAME);
+
```
Comment (yx91490) [inline seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:94]: useless

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        boolean flag = config.hasPath(SparkBatchExecution.SOURCE_TABLE_NAME);
+        Dataset<Row> fromDs;
+        if (flag){
+            String sourceTableName = config.getString(SparkBatchExecution.SOURCE_TABLE_NAME);
+            fromDs = environment.getSparkSession().read().table(sourceTableName);
+        }else {
+            fromDs = ds;
+        }
+
+
```
Comment (yx91490) [inline seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:139]: duplicated blank line

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+            String tableName = config.getString(SparkBatchExecution.RESULT_TABLE_NAME);
+            registerTempView(tableName, ds);
+        }
+    }
+
+    public static void sinkProcess(SparkEnvironment environment, BaseSparkSink sink, Dataset<Row> ds) {
+        Config config = sink.getConfig();
+        boolean flag = config.hasPath(SparkBatchExecution.SOURCE_TABLE_NAME);
+        Dataset<Row> fromDs;
+        if (flag){
```
Comment (yx91490) [inline seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:132]: better rename to a meaningfull variable.

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/BaseSparkBatchSource.java [hunk 7]
```diff
@@ -14,11 +14,12 @@
...
-import io.github.interestinglab.waterdrop.spark.BaseSparkSource
-import org.apache.spark.sql.{Dataset, Row}
+package io.github.interestinglab.waterdrop.spark.batch;
 
-trait SparkBatchSource extends BaseSparkSource[Dataset[Row]]{
+import io.github.interestinglab.waterdrop.spark.BaseSparkSource;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
 
+public abstract class BaseSparkBatchSource extends BaseSparkSource<Dataset<Row>> {
```
[the remark this answers is not in the document]
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/BaseSparkBatchSource.java:24]: In Java code ,the origin name will appear error when complie
image

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/BaseSparkBatchSink.java [hunk 8]
```diff
@@ -14,10 +14,10 @@
...
-package io.github.interestinglab.waterdrop.spark.batch
+package io.github.interestinglab.waterdrop.spark.batch;
 
-import io.github.interestinglab.waterdrop.spark.BaseSparkSink
-
-trait SparkBatchSink extends BaseSparkSink[Unit]{
+import io.github.interestinglab.waterdrop.spark.BaseSparkSink;
+import scala.Unit;
 
+public abstract class BaseSparkBatchSink extends BaseSparkSink<Unit> {
```
[the remark this answers is not in the document]
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/BaseSparkBatchSink.java:22]: same reply :https://github.com/InterestingLab/seatunnel/pull/742#discussion_r771776360

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/BaseSparkSink.java [hunk 9]
```diff
@@ -14,19 +14,30 @@
...
-  override def getConfig: Config = config
+    @Override
+    public void setConfig(Config config){
+        this.config = config;
+    };
+
+    @Override
+    public Config getConfig(){
+        return config;
+    };
```
[the remark this answers is not in the document]
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/BaseSparkSink.java]: already solved

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/BaseSparkSink.java [hunk 10]
```diff
@@ -14,19 +14,30 @@
...
+public abstract class BaseSparkSink<OUT>  implements BaseSink<SparkEnvironment>{
+    protected Config config = ConfigFactory.empty();
 
-  override def setConfig(config: Config): Unit = this.config = config
 
-  override def getConfig: Config = config
+    @Override
+    public void setConfig(Config config){
+        this.config = config;
+    };
```
[the remark this answers is not in the document]
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/BaseSparkSink.java]: already solved

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/BaseSparkSink.java [hunk 11]
```diff
@@ -14,19 +14,30 @@
...
+import io.github.interestinglab.waterdrop.config.Config;
+import io.github.interestinglab.waterdrop.config.ConfigFactory;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
 
-  protected var config: Config = ConfigFactory.empty()
+public abstract class BaseSparkSink<OUT>  implements BaseSink<SparkEnvironment>{
+    protected Config config = ConfigFactory.empty();
 
-  override def setConfig(config: Config): Unit = this.config = config
```
[the remark this answers is not in the document]
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/BaseSparkSink.java:29]: already solved

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/BaseSparkTransform.java [hunk 12]
```diff
@@ -14,20 +14,29 @@
...
 
-  override def getConfig: Config = config
+    @Override
+    public Config getConfig() {
+        return config;
+    }
+
+    public abstract Dataset<Row> process(Dataset<Row>data, SparkEnvironment env);
 
-  def process(data: Dataset[Row], env: SparkEnvironment): Dataset[Row];
```
[the remark this answers is not in the document]
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/BaseSparkTransform.java:41]: already solved

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java (hunk 1 above)
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:39]: already solved

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java (hunk 2 above)
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:47]: already solved

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java (hunk 3 above)
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:95]: already solved

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java (hunk 5 above)
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:139]: already solved

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java (hunk 6 above)
Comment (huzk8) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/SparkBatchExecution.java:132]: already solved

File: seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/BaseSparkBatchSource.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (yx91490) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/BaseSparkBatchSource.java:24]: you can add @SuppressWarnings("PMD.AbstractClassShouldStartWithAbstractNamingRule") on top of class:)
image

[the remark this answers is not in the document]
Comment (yx91490) [reply seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/BaseSparkBatchSource.java:24]: have you noticed this?

**Comment by yx91490** [comment]
ping @huzk8

**Comment by huzk8** [comment]
i will update this pr from latest dev branch later.



发自我的iPhone


------------------ Original ------------------
From: CenterCode ***@***.***&gt;
Date: Wed,Jan 12,2022 10:38 PM
To: apache/incubator-seatunnel ***@***.***&gt;
Cc: huzk ***@***.***&gt;, State change ***@***.***&gt;
Subject: Re: [apache/incubator-seatunnel] [REFACTOR]: change spark batch api code to java (PR #742)





 
@yx91490 commented on this pull request.
 
 
In seatunnel-spark-api/src/main/scala/io/github/interestinglab/waterdrop/spark/batch/BaseSparkBatchSource.java:
 &gt;   +public abstract class BaseSparkBatchSource extends BaseSparkSource<Dataset<Row&gt;&gt; {  
have you noticed this?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by yx91490** [comment]
yes, you can rebase and force push, good luck.

**Pull request closed without merging**